src/error.rs

```diff
@@ -235,7 +244,7 @@ pub(crate) enum Kind {
     Redirect,
     Status(StatusCode),
     Body,
-    Decode,
+    Decode(Option<Bytes>),
```

Comment:
Note that this significantly balloons the size of `Kind` from 4 bytes to _40_ bytes.

Reply:
Yes, you are right! An `Option<Box<Bytes>>` would only balloon from 4 to 12.